packages/gdl-frontend/pages/offline.js

```diff
@@ -63,6 +78,55 @@ class OfflinePage extends React.Component<{}, State> {
     }
   };
 
+  openCloseEditMode = () => {
+    this.setState({
+      editMode: !this.state.editMode,
+      selectedBooks: [],
+      openDialog: false,
+      selectAll: 0
+    });
+  };
+
+  openCloseDialog = () => {
+    if (this.state.selectedBooks.length > 0) {
+      this.setState({ openDialog: !this.state.openDialog });
+    }
+  };
+
+  selectAllBooks = () => {
+    this.state.selectedBooks.length === this.state.books.length
+      ? this.setState({ selectAll: 0, selectedBooks: [] })
+      : this.setState({
+          selectAll: 1,
+          selectedBooks: this.state.books.map(book => book.id)
+        });
+  };
+
+  deselectAllBooks = () => {
+    this.setState({ selectAll: 0, selectedBooks: [] });
+  };
+
+  deleteSelected = async () => {
+    if (!offlineLibrary) return;
+
+    if (this.state.selectedBooks.length !== this.state.books.length) {
+      try {
+        for (let i = 0; i < this.state.selectedBooks.length; i++) {
```

Comment:
Mulig du kan erstatte vanlig for-loop med noe annet? (inspirasjon : https://medium.com/daily-js-tips/replace-your-loops-by-array-methods-map-4e9af2e18427)